generator/restriction_collector.hpp

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#include "indexer/routing.hpp"
+
+#include "std/functional.hpp"
+#include "std/limits.hpp"
+#include "std/string.hpp"
+#include "std/unordered_map.hpp"
+#include "std/utility.hpp"
+#include "std/vector.hpp"
+
+namespace routing
+{
+/// This class collects all relations with type restriction and save feature ids of
+/// their road feature in text file for using later.
+class RestrictionCollector
+{
+  friend void UnitTest_RestrictionTest_ValidCase();
+  friend void UnitTest_RestrictionTest_InvalidCase();
+  friend void UnitTest_RestrictionTest_ParseRestrictions();
+  friend void UnitTest_RestrictionTest_ParseFeatureId2OsmIdsMapping();
+
+public:
+  /// \brief Addresses a link in vector<Restriction>.
+  struct Index
+  {
+    Index(size_t restrictionNumber, size_t linkNumber)
+      : m_restrictionNumber(restrictionNumber), m_linkNumber(linkNumber)
+    {
+    }
+    size_t m_restrictionNumber = 0;  // Restriction number in restriction vector.
+    size_t m_linkNumber =
+        0;  // Link number for a restriction. It's equal to zero or one for most cases.
```

Comment:
The comment is long, move it to a separate line please. Inline comments must be used for inline comments.